src/mondefense.cpp

```diff
@@ -1,38 +1,47 @@
+#include "mondefense.h"
 #include "monster.h"
+#include "creature.h"
+#include "damage.h"
 #include "game.h"
 #include "rng.h"
 #include "line.h"
 #include "bodypart.h"
-#include "speech.h"
 #include "messages.h"
-#include "mondefense.h"
-#include <algorithm>
 
-
-void mdefense::zapback(monster *m, Creature *source, const projectile *proj)
+void mdefense::zapback(monster *const m, Creature *const source, projectile const* const proj)
 {
-    player *foe = dynamic_cast< player* >( source );
-    monster *othermon = dynamic_cast< monster* >( source );
-    if( rl_dist( m->pos(), source->pos() ) > 1 ) {
-        return; // Out of range
+    if ((proj) ||                              // Not a melee attack
+        (rng(0, 100) > m->def_chance) ||       // Attacker lucked out
+        (rl_dist(m->pos(), source->pos()) > 1) // Out of range
+    ) {
+        return;
     }
-    if( proj != nullptr || rng(0, 100) > m->def_chance ) {
-        return; // Not a melee attack or attacker lucked out
+
+    auto const is_eligible_player = [](player const *const foe) {
```

Comment:
I don't have a chip on my shoulder about lambdas, they're fine, the problem
is you trying to use them when there is no benefit.  You're complicating
code for reasons even you admit don't matter.
The rule you're breaking is "do the simplest thing that works".